devito/symbolics/extended_sympy.py

```diff
@@ -537,6 +537,26 @@ class DOUBLEP(CastStar):
 MAX = Function('MAX')
 MIN = Function('MIN')
 
+
+def rfunc(func, item, *args):
+    """
+    A utility function that recursively generates 'func' nested relations.
+
+    Examples
+    ----------
+    >> rfunc(min, [a, b, c, d])
+    MIN(a, MIN(b, MIN(c, d)))
+
+    >> rfunc(max, [a, b, c, d])
+    MAX(a, MAX(b, MAX(c, d)))
+    """
+
+    assert func in rfunc_mapper
+    rf = rfunc_mapper[func]
+
+    return rf(item, rfunc(func, *args)) if len(args) > 0 else item
```

Comment:
we've already replaced most of the ternary operators in devito with explicit if-else conditionals (ie multi-lines)

more lines, but clearer

I used to use ternary operators for conditions too. That was a long time ago. Following the advise of wiser people, I eventually switched to explicit if-else, as I find them more clear.

This doesn't apply to list comprehension and such, of course!

Now, this is just personal taste maybe, and your ternary would be normally OK... but we try to follow conventions in devito, so it would be better if you could switch back to multi-line if-else

Reply:
ok. Thanks for the detailed reasoning.